test/FIFOMarket.test.ts

```diff
@@ -614,6 +614,38 @@ describe('FIFOMarket', () => {
           .toString()
       );
     });
+    it('should sell removals in order of vintage regardless of minting order', async () => {
+      const buyerInitialBPNoriBalance = formatTokenAmount(1_000_000);
+      const { bpNori, fifoMarket, hre, totalAmountOfRemovals } =
+        await setupTestLocal({
+          buyerInitialBPNoriBalance,
+          removalDataToList: [
+            { amount: 1, vintage: 2015 },
+            { amount: 5, vintage: 2014 },
```

Comment:
because `setupTestLocal` uses `batch` for minting in a single call, it might make this test more sound to additionally include a subsequent call to the mint removals without an earlier vintage.